### ISSUE TITLE
[Fix Available 1.309] Surfaces are disappearing

### ISSUE THREAD

**Issue by Hisashimaru** [LogicalError/realtime-CSG-for-unity#42]
How to reproduce
1. Open New Scene
2. Create a brush with box tool
3. Click surface on surface mode

Error log

```
MissingReferenceException: The object of type 'CSGBrush' has been destroyed but you are still trying to access it.
Your script should either check if it is null or you should not destroy the object.
RealtimeCSG.SurfaceEditBrushTool.GetSelectedGameObjects () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/SurfaceEditBrushTool.cs:200)
RealtimeCSG.SurfaceEditBrushTool.UpdateSelection (Boolean allowSubstraction) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/SurfaceEditBrushTool.cs:345)
RealtimeCSG.SurfaceEditBrushTool.HandleEvents () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/SurfaceEditBrushTool.cs:1298)
RealtimeCSG.CSGBrushEditorManager.OnSceneGUI (UnityEditor.SceneView sceneView) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushEditorManager/CSGBrushEditorManager.cs:631)
RealtimeCSG.UnitySceneEventHandler.OnScene (UnityEditor.SceneView sceneView) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/UnitySceneEventHandler.cs:859)
UnityEditor.SceneView.CallOnSceneGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:2046)
UnityEditor.SceneView.HandleSelectionAndOnSceneGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:1381)
UnityEditor.SceneView.OnGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:1218)
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:222)
```

RealtimaCSG 1.308

**Comment by LogicalError** [comment]
Thanks for the bug report, I'll look at it tonight.

**Comment by LogicalError** [comment]
This issue is fixed in version 1.309 which is available on the update site 
( http://prenominal.nl/updates/index.php )
I will close this ticket when it's available on the asset store

**Comment by LogicalError** [comment]
Fix is in the version that's on the asset store now, so closing ticket
